### PR TITLE
Restrict project completion if non terminal engagements

### PR DESCRIPTION
**Pull request by blakekrammes**
Depends on API PR https://github.com/SeedCompany/cord-api-v3/pull/1686

### CODE REVIEW

File: src/scenes/Projects/Update/ProjectWorkflowDialog.tsx [hunk 1]
```diff
@@ -87,9 +87,13 @@ export const ProjectWorkflowDialog = ({
       <Grid container direction="column" spacing={1}>
         {transitions.map((transition, i) => (
           <Tooltip
-            title={`This will change the project step to ${displayProjectStep(
-              transition.to
-            )}`}
+            title={
+              !transition.enabled
+                ? 'The project step cannot be changed to Completed until all engagements have a terminal status'
```
Comment (CarsonF) [inline src/scenes/Projects/Update/ProjectWorkflowDialog.tsx:92]: We should not assume this is the reason the transition is not enabled. The API should provide the reason why it's not enabled.

**Review by CarsonF — CHANGES_REQUESTED** [review]

File: src/scenes/Projects/Update/ProjectWorkflowDialog.tsx (hunk 1 above)
Comment (blakekrammes) [reply src/scenes/Projects/Update/ProjectWorkflowDialog.tsx:92]: What if this is the only one?

Comment (CarsonF) [reply src/scenes/Projects/Update/ProjectWorkflowDialog.tsx:92]: If it feels unnecessary, trust me it's not. There's lots of validation we'd like to add in the future, that's dependent upon the current state. i.e. "the project cannot become active if it doesn't have start or end dates defined."

Comment (blakekrammes) [reply src/scenes/Projects/Update/ProjectWorkflowDialog.tsx:92]: What if we check `enabled` and `to`? Is your main concern keeping the FE clean?

Comment (blakekrammes) [reply src/scenes/Projects/Update/ProjectWorkflowDialog.tsx:92]: Though I guess this does seem like more of an API responsibility.

**Comment by blakekrammes** [comment]
@CarsonF, @sethmcknight  How should we handle engagement statuses if an admin overrides a project to completed? Should that be allowed or prohibited if there are non terminal engagement statuses?

**Comment by CarsonF** [comment]
Overrides are treated like normally workflow, so I would say yes the engagements should be updated.

**Comment by blakekrammes** [comment]
But the normal workflow in this case is to prohibit engagement statuses being updated unless they are individually updated to a terminal status.

**Review by CarsonF — APPROVED** [review]

**Pull request merged**